contracts/BaseLaunchpeg.sol

```diff
@@ -601,4 +494,18 @@ abstract contract BaseLaunchpeg is
             }
         }
     }
+
+    /// @notice Reveals the next batch if the reveal conditions are met
+    function revealNextBatch() external override isEOA {
+        if (!batchReveal.revealNextBatch(totalSupply())) {
+            revert Launchpeg__RevealNextBatchNotAvailable();
+        }
+    }
```

Comment:
Any reason why this is EOA only?

Reply:
Hm this is how it is for our current contracts so I kept it this way

Reply:
That's for when we don't use VRF for the reveal. We kept the possibility to use pseudo-randomness but it actually doesn't make much sense, we're systematically using Chainlink